src/test/test_step.go

```diff
@@ -546,24 +545,15 @@ func moveOutputFile(state *core.BuildState, hash []byte, from, to, dummy string)
 	return fs.RecordAttr(to, hash, xattrName, state.XattrsSupported)
 }
 
-// startTestWorkerIfNeeded starts a worker server if the test needs one.
-func startTestWorkerIfNeeded(tid int, state *core.BuildState, target *core.BuildTarget) error {
-	workerCmd, _, testCmd, err := core.TestWorkerCommand(state, target)
+// verifyWorkerNotNeeded returns an error if a persistent worker is needed.
+func verifyWorkerNotNeeded(tid int, state *core.BuildState, target *core.BuildTarget) error {
+	workerCmd, _, _, err := core.TestWorkerCommand(state, target)
```

Comment:
This eventually calls the sequence replacement stuff for `$(worker)` as well so maybe we can get rid of this?